libraries/AP_Math/AP_Math.cpp

```diff
@@ -128,6 +128,26 @@ float throttle_curve(float thr_mid, float alpha, float thr_in)
     return thr_out;
 }
 
+/*
+ * Convert any base number to any base number. Example octal(8) to decimal(10)
+ * baseIn: base of input number
+ * baseOut: base of output number
+ * inputNumber: value currently in base "baseIn" to be converted to base "baseOut"
+ *
+ * Example: convert ADSB squawk octal "1200" stored in memory as 0x0280 to 0x04B0
+ *          uint16_t squawk_decimal = convertMathBase(8, 10, squawk_octal);
+ */
+uint32_t convertMathBase(const uint8_t baseIn, const uint8_t baseOut, uint32_t inputNumber)
```

Comment:
move to ADSB library, and make specific to octal/decimal. Can we remove the float?

Reply:
float cast removed, hard coded to "10" but there is no integer version of power(). There are other places in the codebase that use ints with powf() though.

Reply:
`pow` is weird:

```
4) Type-generic macro: If any argument has type long double, powl is called. Otherwise, if any argument has integer type or has type double, pow is called. Otherwise, powf is called. If at least one argument is complex or imaginary, then the macro invokes the corresponding complex function (cpowf, cpow, cpowl).
```

(from https://en.cppreference.com/w/c/numeric/math/pow)